### PR TITLE
Add matching service unit tests and GH Actions workflow

### PR DESCRIPTION
**Pull request by TheSpaceCuber**
1. Add mocha tests for all 4 endpoints. But test case coverage is not 100%.
2. Add GitHub actions workflow for matching service.
3. Minor edits to explicitly return status codes in controller.

### CODE REVIEW

**Comment by Herrekt** [comment]
https://stackoverflow.com/questions/28525571/how-to-add-a-delay-before-starting-a-mocha-test-case

Can add hooks that run some delays for test cases that may need to run down the full timer

**Comment by TheSpaceCuber** [comment]
Added return statements 

https://stackoverflow.com/questions/37726863/nodejs-difference-between-res-json-and-return-res-json

File: matching-service/controllers/matchController.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ export const findMatch = async (req, res) => {
                 question: questionResponse.data
             })
             console.log("Interview found for: " + username)
-            res.json({
+            res.status(200).json({
```
Comment (ongweisheng) [inline matching-service/controllers/matchController.js:35]: Add return statement for this res.status(200).json()

**Review by ongweisheng — CHANGES_REQUESTED** [review]
No issues other than the 1 comment and await @hiongkaihan review for merging

**Comment by hiongkaihan** [comment]
LGTM

**Review by hiongkaihan — APPROVED** [review]
LGTM

**Review by ongweisheng — APPROVED** [review]
LGTM!

**Pull request merged**